Enhanced Installation Section in README.md (#35094)
huggingface/transformers: docs/source/en/installation.md

Start by creating a virtual environment in your project directory:

### Create and Activate a Virtual Environment

To avoid dependency conflicts, it’s recommended to use a virtual environment.

#### For macOS/Linux:
```
python3 -m venv env
source env/bin/activate
```
#### For window OS:
```
python -m venv env

> .\env\Scripts\activate

Reviewer: I don't think this section really adds anything new, so I think it can also be removed

Author: Hi @stevhliu,
I have removed the "Create and Activate a Virtual Environment" section as suggested. I understand the concern about redundancy. Would it be helpful to refine this section to emphasize the benefits of virtual environments and include troubleshooting tips? Let me know your thoughts.

See the [How to download files from the Hub](https://huggingface.co/docs/hub/how-to-downstream) section for more details on downloading files stored on the Hub.

</Tip>

> # Troubleshooting

Reviewer: ```suggestion
## Troubleshooting

See below for some of the more common installation issues and how to resolve them.
```

Author: Hi @stevhliu,
Thanks for the suggestion! I have updated the "Troubleshooting" section to a second-level heading and added the introductory line as recommended.

> ## a) Error: Python Version Not Supported
> Ensure you are using Python 3.9 or later.
> To check your Python version, run:
> ```
> python --version
> ```
> ## b) Error: Missing Dependencies
> 
> Install all required dependencies by running:
> ```
> pip install -r requirements.txt
> ```
> Ensure you’re in the project directory before executing the command.
> 
> ## c) Windows-Specific Issues
> 
> If you encounter issues on Windows, you may need to activate Developer Mode. 
> 
> To enable it, navigate to Windows Settings > For Developers > Developer Mode.
> 
> Alternatively, create and activate a virtual environment as shown below:

Reviewer: (Add the code fences around the code snippets, I removed it so the changes are easier to see here)

```suggestion
### Unsupported Python version

Ensure you are using Python 3.9 or later. Run the command below to check your Python version.

python --version

### Missing dependencies

Install all required dependencies by running the following command. Ensure you’re in the project directory before executing the command.

pip install -r requirements.txt

### Windows-specific

If you encounter issues on Windows, you may need to activate Developer Mode. Navigate to Windows Settings > For Developers > Developer Mode.

Alternatively, create and activate a virtual environment as shown below.
```

Author: Hi @stevhliu,
Thanks for the detailed feedback! I have updated the troubleshooting section with simplified headings and proper code fences around the code snippets as suggested.

```
python -m venv env
.\env\Scripts\activate
```

> ### GPU Setup (Optional)
> 
> For GPU acceleration, install the appropriate CUDA drivers for your system:
> 
> a) PyTorch CUDA Compatibility: (https://pytorch.org/get-started/locally)  
> b) TensorFlow GPU Support: (https://www.tensorflow.org/install/pip)  
> 
> #### Verify GPU availability:
> 
> To check if your system detects an NVIDIA GPU, run:

Reviewer: You can include this in the "Install with pip" section as just regular text before the line that says `For CPU-support only`. This will flow better so users don't have to jump to the end to check GPU instructions during installation.

```suggestion
For GPU acceleration, install the appropriate CUDA drivers for [PyTorch](https://pytorch.org/get-started/locally) and TensorFlow(https://www.tensorflow.org/install/pip).

Run the command below to check if your system detects an NVIDIA GPU.
```

Author: Hi @stevhliu,
I have integrated the GPU setup instructions into the "Install with pip" section as regular text and removed the standalone "GPU Setup (Optional)" section.
